### PR TITLE
Add QPU solvers data mocks

### PR DESCRIPTION
**Pull request by randomir**
Prerequisite for #387.

### CODE REVIEW

File: dwave/cloud/testing/mocks.py [hunk 1]
```diff
@@ -135,6 +136,78 @@ def qpu_clique_solver_data(size: int, **kwargs) -> dict:
...
+    qubits = list(graph.nodes)
+    couplers = list(graph.edges)
+    num_qubits = len(qubits)
+
+    params = dict(
+        id=f"dw_{num_qubits}q_mock",
+        description=f"A {num_qubits}-qubit mock QPU solver with chimera topology",
+        qubits=qubits,
+        couplers=couplers,
+        topology={"type": "chimera", "shape": [m, n, t]},
```
Comment (JoelPasvolsky) [inline dwave/cloud/testing/mocks.py:171]: How is the topology propagated through `structured_solver_data`?

File: dwave/cloud/testing/mocks.py [hunk 2]
```diff
@@ -135,6 +136,78 @@ def qpu_clique_solver_data(size: int, **kwargs) -> dict:
...
+        raise RuntimeError("Can't generate Pegasus graph without dwave-networkx. "
+                           "Install with 'dwave-cloud-client[mocks]'.")
+
+    graph = dnx.pegasus_graph(m, fabric_only=False)
+    qubits = list(graph.nodes)
+    couplers = list(graph.edges)
+    num_qubits = len(qubits)
+
+    params = dict(
+        id=f"dw_{num_qubits}q_mock",
```
Comment (JoelPasvolsky) [inline dwave/cloud/testing/mocks.py:200]: Should this be `f"Advantage_{num_qubits}q_mock" ?

File: requirements.txt [hunk 3]
```diff
@@ -13,5 +13,8 @@ dataclasses; python_version == "3.6"
 dimod>=0.10.5
 numpy>=1.16
 
+# for solver mocks
+dwave-networkx>=0.8.9
```
Comment (JoelPasvolsky) [inline requirements.txt:17]: Does this new dependency deserve a comment in reno?

File: dwave/cloud/testing/mocks.py (hunk 2 above)
Comment (randomir) [reply dwave/cloud/testing/mocks.py:200]: It's just a default name, easily overridden with `id` kwarg, but sure, we can make it better.

File: dwave/cloud/testing/mocks.py [hunk 4]
```diff
@@ -135,6 +136,78 @@ def qpu_clique_solver_data(size: int, **kwargs) -> dict:
...
+        **kwargs:
+            Solver properties passed down to :meth:`.structured_solver_data`.
+    """
+    try:
+        import dwave_networkx as dnx
+    except ImportError:     # pragma: no cover
+        raise RuntimeError("Can't generate Pegasus graph without dwave-networkx. "
+                           "Install with 'dwave-cloud-client[mocks]'.")
+
+    graph = dnx.pegasus_graph(m, fabric_only=False)
```
Comment (JoelPasvolsky) [inline dwave/cloud/testing/mocks.py:194]: Why `fabric_only=False` here? Online Advantage QPUs usually have qubits from 30 and up

**Review by JoelPasvolsky — APPROVED** [review]
LGTM

File: requirements.txt (hunk 3 above)
Comment (randomir) [reply requirements.txt:17]: I don't think it does. It's just an optional dependency (like dimod). And we use it just for testing.

File: dwave/cloud/testing/mocks.py (hunk 4 above)
Comment (randomir) [reply dwave/cloud/testing/mocks.py:194]: Good catch. I had it like this initially to match the `num_qubits`, but it makes sense to be closer to an actual online qpu.

File: requirements.txt (hunk 3 above)
Comment (JoelPasvolsky) [reply requirements.txt:17]: We say that to install from source, users should do `pip install -r requirements.txt` and that now has `dwave-networkx` (as well as dimod). How would users know it's optional?

Comment (randomir) [reply requirements.txt:17]: Because it's in package's `mocks` extra. Like dimod is in `bqm`/`dqm` extra. (In setup.py, and declared in package metadata).

Comment (JoelPasvolsky) [reply requirements.txt:17]: Perfect, thanks!

**Pull request merged**